sdk/openai/azure-ai-openai/src/main/java/com/azure/ai/openai/OpenAIClientBuilder.java

```diff
@@ -378,10 +347,12 @@ public OpenAIAsyncClient buildAsyncClient() {
      * @return an instance of OpenAIClient.
      */
     public OpenAIClient buildClient() {
-        if (nonAzureOpenAIKeyCredential != null) {
-            return new OpenAIClient(buildInnerNonAzureOpenAIClient());
+        // Connect to OpenAI service needs to have endpoints start with "https://api.openai.com/".
+        // By default, if user doesn't provide endpoint, it will redirect to create an Azure OpenAI service.
+        if (endpoint != null && !endpoint.startsWith(OPEN_AI_ENDPOINT)) {
+            return new OpenAIClient(buildInnerClient());
```

Comment:
Instead of repeating this code, we should extract this to a method `useAzureOpenAI()`.